### PR TITLE
4222/allow new case without court date

### LINKED ISSUE #4222: Allow case creation without next court report date BUT only with clicking confirm button
**What type(s) of user does this feature affect?**
 - supervisors
 - admins

**Description**
In the event that the court date for a new case being created is not known, currently we block creation of the case. We don't want to do that. So we want to add a button that says "I don't have it at the moment" to the court date part of the new case creation workflow. This button should allow the case to be created without a court date. 

**Screenshots of current behavior, if any**  
You can paste images on the clipboard here

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by andreLumor**
### What github issue is this PR for, if any?
Resolves #4222

### What changed, and why?
Created a checkbox that hides "next court date" input, allowing casa case to be created without it.

### How will this affect user permissions?
- Admin permissions: now are able to create casa case without next court date

### How is this tested? (please write tests!) 💖💪
to execute tests, please run:
`rspec spec/system/casa_cases/new_spec.rb:79`

### Screenshots please :)
[Screencast from 23-12-2022 16:20:51.webm](https://user-images.githubusercontent.com/36737050/209396815-f42696bd-fbdb-40af-a1df-022739856c4a.webm)

### CODE REVIEW

File: app/controllers/casa_cases_controller.rb [hunk 1]
```diff
@@ -40,8 +40,14 @@ def edit
   end
 
   def create
+    create_params = if casa_case_params[:empty_court_date] == "1"
```
Comment (littleforest) [inline app/controllers/casa_cases_controller.rb:43]: How about creating a private method here that describes the condition, like:

```
  def create
    create_params = if court_date_unknown?
    ...
    end
  end

...


  private

  def court_date_unknown?
    casa_case_params[:empty_court_date] == "1"
  end
```

Then it is a little bit easier to read the code like a sentence and understand what is happening.

File: app/views/casa_cases/_form.html.erb [hunk 2]
```diff
@@ -60,10 +60,11 @@
             <div class="field form-group">
               <%= cdf.text_field :date,
                                  data: {provide: "datepicker", date_format: "yyyy/mm/dd"},
-                                 class: "form-control",
-                                 :required => 'required' %>
+                                 class: "form-control" %>
             </div>
         <% end %>
+        <%= form.check_box :empty_court_date, class: "toggle-court-date-input" %>
+        <%= form.label :empty_court_date, "I don't have it at the moment" %>
```
Comment (littleforest) [inline app/views/casa_cases/_form.html.erb:67]: I know the ticket issue gave the "I don't have it at the moment", but I think the "it" is a little confusing to what it is referring to. Can we change it to: 

```suggestion
        <%= form.label :empty_court_date, "I don't know the court date yet" %>
```

File: spec/system/court_dates/new_spec.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@
   before do
     travel_to now
     sign_in admin
-    visit casa_case_path(casa_case)
+    visit edit_casa_case_path(casa_case)
```
Comment (littleforest) [inline spec/system/court_dates/new_spec.rb:18]: Is there a reason why this had to be changed?

**Review by littleforest — COMMENTED** [review]
@andreLumor this looks great! Just some minor suggestions.

**Review by compwron — APPROVED** [review]

File: spec/system/court_dates/new_spec.rb (hunk 3 above)
Comment (compwron) [reply spec/system/court_dates/new_spec.rb:18]: This seems strange, what's up here?

Comment (andreLumor) [reply spec/system/court_dates/new_spec.rb:18]: "add court date" button in casa_case_path isn't being rendered, so I assumed it was referring to the wrong page. I'll discover why it isn't rendering

Comment (andreLumor) [reply spec/system/court_dates/new_spec.rb:18]: My bad, the css wasn't compiling correctly. Already corrected it.

File: app/views/casa_cases/_form.html.erb (hunk 2 above)
Comment (andreLumor) [reply app/views/casa_cases/_form.html.erb:67]: I'll change it.

File: app/controllers/casa_cases_controller.rb (hunk 1 above)
Comment (andreLumor) [reply app/controllers/casa_cases_controller.rb:43]: Agreed! I refactored it, creating a casa_case_create_params, so the create method is cleaner now.

**Review by littleforest — APPROVED** [review]
🎉

**Pull request merged**